include/deal.II/lac/scalapack.h

```diff
@@ -180,21 +180,106 @@ class ScaLAPACKMatrix : protected TransposeTable<NumberType>
+  * \f$U \in \mathbb{R}^{M \times M}\f$ and \f$U \in \mathbb{R}^{M \times M}\f$
+  * as orthogonal matrices. The diagonal elements of \f$\Sigma\f$
+  * are the singular values of \f$A\f$ and the columns of \f$U\f$ and \f$V\f$ are the
+  * corresponding left and right singular vectors, respectively. The
+  * singular values are returned in decreasing order and only the first \f$\min(M,N)\f$
+  * columns of \f$U\f$ and rows of VT = \f$V^T\f$ are computed.
+  *
+  * Upon return the content of the matrix is unusable.
+  * The matrix A must have identical block cyclic distribution for the rows and column.
+  *
+  * If left singular vectors are required matrices \f$A\f$ and \f$U\f$
+  * have to be constructed with the same process grid and block cyclic distribution.
+  * If right singular vectors are required matrices \f$A\f$ and \f$V^T\f$
+  * have to be constructed with the same process grid  and block cyclic distribution.
+  *
+  * To avoid computing the left and/or right singular vectors the function accepts <code>nullptr</code>
+  * for @p U and/or @p VT.
+  */
+  std::vector<NumberType> compute_SVD(ScaLAPACKMatrix<NumberType> *U = nullptr,
+                                      ScaLAPACKMatrix<NumberType> *VT = nullptr);
+
+
+
+  /**
+  * Solving overdetermined or underdetermined real linear
+  * systems involving matrix \f$A \in \mathbb{R}^{M \times N}\f$, or its transpose \f$A^T\f$,
+  * using a QR or LQ factorization of \f$A\f$ for \f$\mbox{NRHS}\f$ RHS vectors in the columns of matrix \f$B\f$
+  *
+  * It is assumed that \f$A\f$ has full rank: \f$rank(A) = \min(M,N)\f$.
+  *
+  * The following options are supported:
+  * - 1. If <code>transpose==false</code> and \f$M \geq N\f$: least squares solution of overdetermined system
+  *      \f$\min \Vert B - A*X\Vert\f$.
+  *
+  *      Upon exit the rows 0 to N-1 of \f$B\f$ contain the least square solution vectors. The residual sum of squares
+  *      for each column is given by the sum of squares of elements N to M-1 in that column
+  *
+  * - 2. If <code>transpose==false</code> and \f$M < N\f$: find minimum norm solutions of underdetermined systems
+  *      \f$A * X = B\f$.
+  *
+  *      Upon exit the columns of \f$B\f$ contain the minimum norm solution vectors
+  *
+  * - 3. If <code>transpose==true</code> and and \f$M \geq N\f$: find minimum norm solutions of underdetermined system
+  *      \f$ A^\top X = B\f$
+  *
+  *      Upon exit the columns of \f$B\f$ contain the minimum norm solution vectors
+  *
+  * - 4. If <code>transpose==true</code> and \f$M < N\f$: least squares solution of overdetermined system
+  *      \f$\min \Vert B - A^\top X\Vert\f$.
+  *
+  *      Upon exit the rows 0 to M-1 contain the least square solution vectors. The residual sum of squares
+  *      for each column is given by the sum of squares of elements M to N-1 in that column
+  * .
+  * If <code>transpose==false</code> it is \f$B\f \in \mathbb{R}^{M \times \mbox{NRHS}}$,
```

Comment:
there is a bug in the documentation (`\f`), i would also write it `B \in \mathbb{R}^{M \times N_{\rm RHS}}` and adjust the usage of `NRHS` elsewhere.

Also probably you wanted to write:

If ...  then \<formula\>